README.md
# HXLTM - ontologia and initial reference implementation
[![Site](https://img.shields.io/badge/Site-hxltm.etica.ai-blue[Site] "Site")](https://hxltm.etica.ai/)
[![GitHub](https://img.shields.io/badge/GitHub-EticaAI%2Fhxltm-lightgrey?logo=github&style=social[EticaAI/hxltm] "GitHub")](https://github.com/EticaAI/hxltm)
[![Pypi: hxltm-eticaai](https://img.shields.io/badge/python%20pypi-hxltm--eticaai-brightgreen[Python] 
 "Pypi: hxltm-eticaai")](https://pypi.org/project/hxltm-eticaai/)

The _Multilingual Terminology in Humanitarian Language Exchange_
(abbreviation: _HXLTM_) is an HXLated valid tabular format (stricly, a
documented subset of [HXL Standard](https://hxlstandard.org/[HXL]) started by
[HXL-CPLP](https://github.com/HXL-CPLP)) with strong focus to store community
contributed translations and multilingual terminology while maximizing
portability for implementers.

**The documentation is available at <https://hxltm.etica.ai/>.**

```bash
pip install hxltm-eticaai --upgrade
```


<!--
---

Multilingual Terminology in Humanitarian Language Exchange (HXLTM) - Ontologia and reference implementation

Quickstart

- <https://hxltm.etica.ai/>

```bash
pip install hxltm-eticaai
```

---

(like via use of `+hxltmcli+` to export to
Excel, +++<abbr title="XML Localization Interchange File Format">XLIFF</abbr>+++,
+++<abbr title="Translation Memory eXchange">TMX</abbr>+++,
+++<abbr title="TermBase eXchange">TBX</abbr>+++,
+++<abbr title="Universal Terminology eXchange">UTX</abbr>+++, JSON, CSV,
link:archivum/[and more!]



**[monorepo split not finished] See https://github.com/EticaAI/HXL-Data-Science-file-formats/issues/25**

- <https://hxltm.etica.ai/>

```bash
pip install hxltm-eticaai
```
-->

<!--

> TODO:
> - https://github.com/darshandsoni/asciidoctor-skins
> - https://github.com/coveooss/json-schema-for-humans
> - https://github.com/ymyzk/tox-gh-actions
> - re-add UNLICENSE
> - re-add dedicated tests with tox
> - documentation with one of the following:
>   - https://docs.antora.org/ / https://antora.org/
>   - https://github.com/manoelcampos/asciidoctor-ghpages-action
> - documentation that could be localized?
>   - https://github.com/OpenRefine/OpenRefine/issues/2273#issuecomment-574415046

- https://hxltm.etica.ai
- https://eticaai.github.io/hxltm/

-->

## Licenses

> Public domain means each issue only needs to be resolved once

## Software license

[![Public Domain](https://i.creativecommons.org/p/zero/1.0/88x31.png)](UNLICENSE)

To the extent possible under law, [Etica.AI](https://github.com/EticaAI)
and non anonymous collaborators have waived all copyright and related or
neighboring rights to this work to [Public Domain](UNLICENSE).

Optionally, the [BSD Zero Clause License](https://spdx.org/licenses/0BSD.html)
is also one explicit alternative to the Unlicense as an older license approved
by the Open Source Initiative:

`SPDX-License-Identifier: Unlicense OR 0BSD`

## Creative content license (algorithms and concepts as pivot exchange for other data standards, and user documentation)

[![Public Domain](https://i.creativecommons.org/p/zero/1.0/88x31.png)](UNLICENSE)

To the extent possible under law, [Etica.AI](https://github.com/EticaAI)
and non anonymous collaborators have waived all copyright and related or
neighboring rights to this work to public domain dedication. As 2021, the
[CC0 1.0 Universal (CC0 1.0) Public Domain Dedication](https://creativecommons.org/publicdomain/zero/1.0/)
with **additional right** to you granted upfront:

- The collaborators explicitly waive any potential **patent rights** for
  algorithms/ideas. We also preemptively ask that potential requests from our
  heirs in unforeseen future in any jurisdiction be ignored by any regional or
  international court.

<details>
<summary>More context about this</summary>

This different license for creative content is mostly for lawyers who would
complain about use of Unlicense for creative content. More context (from the
point of open source) on waiving patent rigths explicitly (since no better license for
creative content do exist) is here: <https://opensource.org/faq#cc-zero>.

There is no interest by ourselves to do _patent troll_ (for monetary gain)
or allow abuse copyrights (to enforce companies, organizations, o
governments) even if:

- we directly strongly disagree
- some entity try to use us as proxy to enforce us some sort of boycott to any
  other entity.

Note that data exchange on humanitarian context, even outside global-like
war-time, already is quite complex and the need of accurate linguistic content
conversion still even more critical to not have know errors. While the idea of
stories behind cases like the "_黙殺_" ("_mokusatsu_") are disputable, the
modern tooling to deal with multilingual terminology (including used to
create dictionaries) is prone to human error.

</details>

<!--
  ## Licenses for trademarks, terms, etc
  ### Comments
-->

CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## [Unreleased]

### Added
- hxltmcli now implement libhxl `--expand-merged` option
- hxltmcli now implement libhxl `--scan-ckan-resources` option

### Changed
- #14 updated to be compatible with libhxl 4.25.1. **Note: previous versions
  are know to not work if used with libhxl newer than 4.25 (which is default
  behavior for newer installations)**
- Upgraded to libhxl 4.25.2

### Removed
- #14 removed option to run with libhxl < 4.25.1. Behavior: raises exception
  documenting version compatibilities.

## [v0.9.1] - 2022-06-22

> Note: this version may require you to install additional dependencies. Varies
        by cli tool you plan to install. Can do all then with
        `pip install libhxl pyyaml defusedxml langcodes python-liquid`

### Added
- Improved documentation

### Fixed
- Changed references from `hxl.io` to `hxl.input` to work with lastest libhxl
  version
- Fixed issue with unpublished RC version of `hxltmcli` being released as
  `hxltmdexml`

## [v0.8.10] - 2020-11-26

### Added
- tmeta files `.tmeta.215.yml` (cli option --tmeta-archivum) now expose
  variables from script neutral `____` (four underscores; allusion to
  _breaking the forth wall_, give extra metadata)

### Changed
- Generate XML now have less white spaces
- `hxltmdexml` now also export HXLTM with deterministic ordering (based on `--agendum-linguam`)

### Fixed
- Bugfixed issue with importing back term value (only on the generic XML
  importer; TMX/TBX was not affected)
- Bugfixed not previously detected corner case of not exporting exactly the
  first item (non HXLTM formats, and roundtrip)
- Tags like Ancient Greek, grc-Grek, should still have BCP47 (simplify templating)

## [v0.8.9] - 2020-11-17
### Added
- #8 Official releases now will include PDF documentation 
- #7 `ontologia/0.77.995.yml` added (but not implemented) 
- #7 `ontologia/cor.hxltm.215.json` and added `ontologia/0.77.995.json`
- `hxltmcli --versionem`
- `hxltmdexml --versionem`

### Changed
- #7 `ontologia/cor.hxltm.215.yml` can be use instead of `ontologia/cor.hxltm.yml` (refs #7)
- #7 Environment variable `HXLM_CONFIG_BASE` renamed to `HXLTM_DORMUM`
- #7 Environment variable `HXLTM_TESTUM_BASIM_DEFALLO` renamed to `HXLTM_TESTUM`
- If no `cor.hxltm.215.yml`, now exception explain search paths as a hint.

## [v0.8.9-rc.2] - 2020-11-14

### Added
- <https://hxltm.etica.ai>, working draft of new documentation page. Based on (eventually to be deprecated) <https://hdp.etica.ai/hxltm/>
- `testum/de-python/testinfra/test_simplex.py` de `EticaAI/HXL-Data-Science-file-formats/tests/test_shell_basic.py` 

### Changed
- Move: `testum/hxltm/*` -> `testum/`
- Move: `docs/*` -> `docs/eng-Latn/*`: now the documentation page have explicitly
  one language, so it _could_ be translated eventually
- Tox tests pass again. Required minor changes on search path for test dataset
  without need to specify environment variables (e.g. `testum/hxltm` to
  `testum/hxltm` as base). Still need some clean up.

### Fixed
- `bin/hxltmdexml,py` -> `bin/hxltmdexml.py`


## [v0.8.9-rc.1] - 2020-11-12

### Added
- `bin/hxltmcli.py` v0.8.8 from `EticaAI/HXL-Data-Science-file-formats/bin/hxltmcli` hdp-toolchain v0.8.8.6.post1
- `bin/hxltmdexml.py` v0.8.0 from `EticaAI/HXL-Data-Science-file-formats/bin/hxltmdexml` hdp-toolchain v0.8.8.6.post1
- `ontologia/cor.hxltm.yml` v0.8.6+EticaAI+voluntārium-commūne from `EticaAI/HXL-Data-Science-file-formats/bin/hxltmdexml` hdp-toolchain v0.8.8.6.post1
- `testum/hxltm/` from `EticaAI/HXL-Data-Science-file-formats/testum/hxltm/` hdp-toolchain v0.8.8.6.post1
- `docs/` from `EticaAI/HXL-Data-Science-file-formats/hxltm/` hdp-toolchain v0.8.8.6.post1

### Changed
- Started monorepo split from <https://github.com/EticaAI/HXL-Data-Science-file-formats>

## [v0.8.8] - 2020-11-07

> NOTE: since the monorepo EticaAI/HXL-Data-Science-file-formats did not
> have a CHANGELOG.md, even if changelogs generated by git log are not
> good practice, for historical proposes, this new CHANGELOG.md on this
> dedicated repository will have the old changes listed. The
> Command used to generate it was:
>
> `/workspace/git/EticaAI/HXL-Data-Science-file-formats$ git log --pretty=format:'%h %ad %s (%an)' --date=short -- ontologia/cor.hxltm.yml bin/hxltmcli bin/hxltmdexml testum/hxltm/`

```log
c7c5aa5 2021-11-07 hdp-toolchain v0.8.8.6.post1; hxmlcli vv0.8.8: --objectivum-formulam, allow save result to file (was working for stdout) (Emerson Rocha)
d347587 2021-11-06 hdp-toolchain v0.8.8.5; reduce dependencies for hxltmcli/hxltmdexml (refs fititnt/hxltm-action#1) (Emerson Rocha)
a18d179 2021-11-06 hxltm: latin name from 'Terminologium Multilinguam' to 'Terminologia Multilinguae'; still need to review many other terms (Emerson Rocha)
f4fe986 2021-10-19 hxltmcli (#22): documented ontologia:normam.HXLTM general notes (Emerson Rocha)
64f8f2c 2021-10-19 hxltmcli (#22): documented HXL-Proxy as data source (Emerson Rocha)
87a7a75 2021-10-19 hxltmcli (#22): improved more documentation (Emerson Rocha)
476e4fd 2021-10-19 hxltmcli (#22): improved documentation (Emerson Rocha)
3b75806 2021-10-19 hxltmcli (#24): --tmeta-archivum; Terminologium Multilinguam (priore HXL Trānslātiōnem Memoriam) (Emerson Rocha)
50b21f9 2021-10-19 hxltmcli (#24): --meta-archivum (Emerson Rocha)
45cbf2a 2021-10-19 hxltmcli (#24): started --meta parameter; HXLTM additional file for advanced processing (YAML-like, as opposed to table-like) (Emerson Rocha)
f4b6ad6 2021-10-15 hdp-toolchain v0.8.8.4 released on pypi.org; hxltmcli (#58) v0.8.7: MVP of --objectivum-formulam (Emerson Rocha)
697aa49 2021-10-15 hxltmcli (#23): HXLTMDatumNormam.aequale() (Emerson Rocha)

testum/de-python/testinfra/test_simplex.py
#!/usr/bin/env python3

# To output even more verbose results
#     ./testum/de-python/testinfra/test_simplex.py
#
# To test directly
#     pytest -vv ./testum/de-python/testinfra/test_simplex.py

# NOTE: this file require testinfra, https://testinfra.readthedocs.io/en/latest/
#     pip3 install pytest-testinfra

# TODO: https://testinfra.readthedocs.io/en/latest/examples.html
#       Implement things via parameters


# @see https://github.com/amoffat/sh
# @see https://amoffat.github.io/sh/
# @see https://blog.esciencecenter.nl/testing-shell-commands-from-python-2a2ec87ebf71
# @see https://github.com/NLeSC/python-template/blob/master/tests/test_project.py

import os
import sys
import pytest

# import pytest.testinfra

# import yaml


# print(sys.version_info)
# print(sys.version_info <= (3, 9))
# print(sys.version_info == (3, 8))

def test_hxltmdexml_help(host):
    cmd = host.run("hxltmdexml --help")

    assert cmd.succeeded


def test_hxltmcli_help(host):
    cmd = host.run("hxltmcli --help")

    assert cmd.succeeded


def test_hxltmdexml_versionem(host):
    cmd = host.run("hxltmdexml --versionem")

    assert cmd.succeeded


def test_hxltmcli_versionem(host):
    cmd = host.run("hxltmcli --versionem")

    assert cmd.succeeded


# def test_libhxl_help(host):
#     cmd = host.run("hxlspec --help")

#     assert cmd.succeeded


# def test_hdpcli_help(host):
#     cmd = host.run("hdpcli --help")

#     assert cmd.succeeded


# def test_hxltmcli_help(host):
#     cmd = host.run("hxltmcli --help")

#     assert cmd.succeeded


# def test_urnresolver_help(host):
#     cmd = host.run("urnresolver --help")

#     assert cmd.succeeded


# def test_hxl2tab_help(host):
#     cmd = host.run("hxl2tab --help")


# def test_hxlquickimport_help(host):
#     cmd = host.run("hxlquickimport --help")

#     assert cmd.succeeded


# def test_hxlquickmeta_help(host):
#     # Orange3 dependency (Qt) seems to broke hxlquickmeta on py37
#     # So is not installed. That's why the extra check.

#     if host.exists('hxlquickmeta') and \
#             host.package('Orange3').is_installed and \
#             host.package('pandas').is_installed:

#         cmd = host.run("hxlquickmeta --help")
#         assert cmd.succeeded

# # # Returns empty:
# # hdpcli tests/hrecipe/salve-mundi.hrecipe.mul.hdp.yml --non-grupum salve-mundi


# def test_hdpcli_errors(host):
#     cmd = host.run(
#         "hdpcli tests/hrecipe/FILE_THAT_DO_NOT_EXIST.mul.hdp.yml --non-grupum salve-mundi")

#     # assert not cmd.succeeded
#     assert cmd.stderr != ''
#     assert cmd.rc != 0


# def test_hdpcli_empty(host):
#     cmd = host.run(
#         "hdpcli tests/hrecipe/salve-mundi.hrecipe.mul.hdp.yml --non-grupum salve-mundi")

#     assert cmd.succeeded
#     assert cmd.stderr == ''
#     assert cmd.rc == 0

# # TODO: more specific tests need to be done with actuall results of hdpcli.
# #       It may be easy to do it with the final cli than only with internal
# #       library


testum/de-python/testinfra/test_disciplinam_manuale.py
#!/usr/bin/env python3

# testum/de-python/testinfra/test_disciplinam_manuale.py is a port of
# testum/disciplinam-manuale-anglicum.sh

# To output even more verbose results
#     ./testum/de-python/testinfra/test_disciplinam_manuale.py
#
# To test directly
#     pytest -vv ./testum/de-python/testinfra/test_disciplinam_manuale.py

# NOTE: this file require testinfra, https://testinfra.readthedocs.io/en/latest/
#     pip3 install pytest-testinfra

# TODO: https://testinfra.readthedocs.io/en/latest/examples.html
#       Implement things via parameters


# @see https://github.com/amoffat/sh
# @see https://amoffat.github.io/sh/
# @see https://blog.esciencecenter.nl/testing-shell-commands-from-python-2a2ec87ebf71
# @see https://github.com/NLeSC/python-template/blob/master/tests/test_project.py

import os
import sys
import pytest

# import pytest.testinfra

# TODO: fix this on tox
def test_disciplinam_manuale_hxltmcli_hxltm_csv(host):
    # host.run('cd testum/')
    # cmd = host.run("hxltmcli hxltm-exemplum-linguam.tm.hxl.csv")
    # cmd = host.run("cd testum/; ls -l")
    # cmd = host.run("cd testum/; ls")
    cmd = host.run("hxltmcli --help")
    # cmd = host.run("pwd")
    # cmd = host.run("hxltmcli testum/hxltm-exemplum-linguam.tm.hxl.csv")
    # cmd2 = host.run("cat testum/hxltm-exemplum-linguam.tm.hxl.csv | hxltmcli")
    # cmd = host.run("pwd")

    assert cmd.succeeded
    # assert cmd2.succeeded

    # assert print(cmd)

# TODO: HXLTM-ASA


# def test_disciplinam_manuale_hxltmcli_csv_3(host):
#     # host.run('cd testum/')
#     # cmd = host.run("hxltmcli hxltm-exemplum-linguam.tm.hxl.csv")
#     # cmd = host.run("cd testum/; ls -l")
#     cmd = host.run(
#         "cd testum/; hxltmcli hxltm-exemplum-linguam.tm.hxl.csv --objectivum-CSV-3")
#     # cmd = host.run(
#     #     "cd testum/; hxltmcli hxltm-exemplum-linguam.tm.hxl.csv --objectivum-CSV-3 --fontem-linguam por-Latn@pt --objectivum-linguam spa-Latn@es")
#     # cmd = host.run("pwd")

#     assert cmd.succeeded


testum/manuale-testum.sh
#!/bin/sh
#===============================================================================
#
#          FILE:  manuale-testum.sh
#
#         USAGE:  ./testum/manuale-testum.sh
#
#   DESCRIPTION:  Manual testum for hxltmcli.
#                 This file can also be used to undestand how the output
#                 of testum/hxltm/resultatum was created
#                 DEPRECATED. Use the disciplinam-manuale-anglicum.sh
#                 (both used for testing and generate documentation)
#
#       OPTIONS:  ---
#
#  REQUIREMENTS:  - ./testum/download-hxltm-datum.sh
#          BUGS:  ---
#         NOTES:  ---
#        AUTHOR:  Emerson Rocha <rocha[at]ieee.org>
#       COMPANY:  EticaAI
#       LICENSE:  Public Domain dedication
#                 SPDX-License-Identifier: Unlicense
#       VERSION:  v1.1
#       CREATED:  2021-07-03 03:08 UTC v1.0
#      REVISION:  2021-11-13 23:21 UTC v1.1 moved from
#                                      EticaAI/HXL-Data-Science-file-formats to
#                                      EticaAI/hxltm. Some paths changes
#===============================================================================
# Comment next line if not want to stop on first error
# set -e

echo "DEPRECATED"
exit 0

#### Variables ________________________________________________________________
ROOTDIR="$(pwd)"

# For HXLTM-Exemplum
# @see https://docs.google.com/spreadsheets/d/1isOgjeRJw__nky-YY-IR_EAZqLI6xQ96DKbD4tf0ZO8/edit#gid=0

HXLTM_EXEMPLUM_XLSX_LOCAL="HXLTM-Exemplum.xlsx"
HXLTM_EXEMPLUM_XLSX_GSHEETS="https://docs.google.com/spreadsheets/d/1isOgjeRJw__nky-YY-IR_EAZqLI6xQ96DKbD4tf0ZO8/export?format=xlsx"

# HXLTM_EXEMPLUM_LINGUAM_LOCAL="${ROOTDIR}/hxltm-linguam.tm.hxl.csv"
HXLTM_EXEMPLUM_LINGUAM_LOCAL="hxltm-exemplum-linguam.tm.hxl.csv"
HXLTM_EXEMPLUM_LINGUAM_LOCAL_TMETA="hxltm-exemplum-linguam.tmeta.yml"
HXLTM_EXEMPLUM_LINGUAM_GSHEETS="https://docs.google.com/spreadsheets/d/1isOgjeRJw__nky-YY-IR_EAZqLI6xQ96DKbD4tf0ZO8/edit#gid=1241276648"

HXLTM_EXEMPLUM_LINGUAM_RESULTATUM_TMX_LOCAL="resultatum/hxltm-exemplum-linguam.tmx"
HXLTM_EXEMPLUM_LINGUAM_RESULTATUM_TBX_LOCAL="resultatum/hxltm-exemplum-linguam.tbx"
HXLTM_EXEMPLUM_LINGUAM_RESULTATUM_XLIFF_LOCAL="resultatum/hxltm-exemplum-linguam.xlf"

# .gitignore: Do not save production test files to save space outside hapi.etica.ai
# Hapi_schemam_un_htcds
Hapi_schemam_un_htcds="schemam-un-htcds.tm.hxl.csv"
Hapi_schemam_un_htcds_RESULTATUM_TMX_LOCAL="resultatum/schemam-un-htcds.tmx"
Hapi_schemam_un_htcds_RESULTATUM_XLIFF_LOCAL="resultatum/schemam-un-htcds.xlf"
Hapi_schemam_un_htcds_RESULTATUM_BILINGUAL_HXL_CSV_LOCAL="resultatum/schemam-un-htcds_eng-Latn_por-Latn.hxl.csv"

#### Init testum and chechs _____________________________________________________
cd "$ROOTDIR/testum" || exit

if [ ! -f "$HXLTM_EXEMPLUM_XLSX_LOCAL" ]; then
    echo "$HXLTM_EXEMPLUM_XLSX_LOCAL not found."
    echo "$HXLTM_EXEMPLUM_XLSX_GSHEETS download now..."
    wget -qO- "$HXLTM_EXEMPLUM_XLSX_GSHEETS" > "$HXLTM_EXEMPLUM_XLSX_LOCAL"
# else
    #echo "$HXLTM_EXEMPLUM_XLSX_LOCAL exists"
fi

#### main ______________________________________________________________________

printf "\n\n\n\tTESTUM 001 HXLTM_EXEMPLUM_LINGUAM_GSHEETS\n\n"
echo "hxltmcli $HXLTM_EXEMPLUM_LINGUAM_GSHEETS | grep L10N_ego_codicem"
hxltmcli "$HXLTM_EXEMPLUM_LINGUAM_GSHEETS" | grep L10N_ego_codicem

printf "\n\n\n\tTESTUM 002 HXLTM_EXEMPLUM_LINGUAM_LOCAL\n\n"
echo "hxltmcli $HXLTM_EXEMPLUM_LINGUAM_LOCAL | grep L10N_ego_codicem"
hxltmcli "$HXLTM_EXEMPLUM_LINGUAM_LOCAL" | grep L10N_ego_codicem

printf "\n\n\n\tTESTUM 003 HXLTM_EXEMPLUM_XLSX_LOCAL\n\n"
echo "hxltmcli --sheet 2 $HXLTM_EXEMPLUM_XLSX_LOCAL | grep L10N_ego_codicem" 
hxltmcli --sheet 2 "$HXLTM_EXEMPLUM_XLSX_LOCAL" | grep L10N_ego_codicem

printf "\n\n\n\tTESTUM 004 HXLTM_EXEMPLUM_LINGUAM_RESULTATUM_TMX_LOCAL\n\n"
echo hxltmcli "$HXLTM_EXEMPLUM_LINGUAM_LOCAL" "$HXLTM_EXEMPLUM_LINGUAM_RESULTATUM_TMX_LOCAL" --objectivum-TMX
hxltmcli "$HXLTM_EXEMPLUM_LINGUAM_LOCAL" "$HXLTM_EXEMPLUM_LINGUAM_RESULTATUM_TMX_LOCAL" --objectivum-TMX

printf "\n\n\n\tTESTUM 004 HXLTM_EXEMPLUM_LINGUAM_RESULTATUM_TMX_LOCAL\n\n"
echo hxltmcli "$HXLTM_EXEMPLUM_LINGUAM_LOCAL" "$HXLTM_EXEMPLUM_LINGUAM_RESULTATUM_TBX_LOCAL" --objectivum-TBX-Basim --tmeta-archivum "$HXLTM_EXEMPLUM_LINGUAM_LOCAL_TMETA"
hxltmcli "$HXLTM_EXEMPLUM_LINGUAM_LOCAL" "$HXLTM_EXEMPLUM_LINGUAM_RESULTATUM_TBX_LOCAL" --objectivum-TBX-Basim

printf "\n\n\n\tTESTUM 006 HXLTM_EXEMPLUM_LINGUAM_RESULTATUM_XLIFF_LOCAL\n\n"
echo hxltmcli "$HXLTM_EXEMPLUM_LINGUAM_LOCAL" "$HXLTM_EXEMPLUM_LINGUAM_RESULTATUM_XLIFF_LOCAL" --objectivum-XLIFF
hxltmcli "$HXLTM_EXEMPLUM_LINGUAM_LOCAL" "$HXLTM_EXEMPLUM_LINGUAM_RESULTATUM_XLIFF_LOCAL" --objectivum-XLIFF

# hxltmcli hxltm-exemplum-glossarium-minimum.tm.hxl.csv

## The hxltmcli fails with non HXLated input
#     hxltmcli csv-3-exemplum.csv
## This is one way to prepare the input using hxltag (libhxl-python tool)
#     hxltag -m en-GB#item+rem+i_en+i_eng+is_latn -m pt-PT#item+rem+i_pt+i_por+is_latn -m Comment#meta csv-3-exemplum.csv
#     hxltag -m en-GB#item+rem+i_en+i_eng+is_latn -m pt-PT#item+rem+i_pt+i_por+is_latn -m Comment#meta csv-3-exemplum.csv | hxltmcli
## ... --JSON-kv
#     hxltag -m en-GB#item+rem+i_en+i_eng+is_latn -m pt-PT#item+rem+i_pt+i_por+is_latn -m Comment#meta csv-3-exemplum.csv | hxltmcli -f eng-Latn@en-GB -o por-Latn@pt-PT --JSON-kv
#     hxltag -m en-GB#item+rem+i_en+i_eng+is_latn -m pt-PT#item+rem+i_pt+i_por+is_latn -m Comment#meta csv-3-exemplum.csv | hxltmcli -f eng-Latn@en-GB -o por-Latn@pt-PT --JSON-kv > resultatum/json/pt.csv
## ... --CSV-3
#     hxltag -m en-GB#item+rem+i_en+i_eng+is_latn -m pt-PT#item+rem+i_pt+i_por+is_latn -m Comment#meta csv-3-exemplum.csv | hxltmcli -f eng-Latn@en-GB -o por-Latn@pt-PT --CSV-3
#     hxltag -m en-GB#item+rem+i_en+i_eng+is_latn -m pt-PT#item+rem+i_pt+i_por+is_latn -m Comment#meta csv-3-exemplum.csv | hxltmcli -f eng-Latn@en-GB -o por-Latn@pt-PT --CSV-3 > resultatum/csv-3-exemplum.csv

printf "\n\n\n\tTESTUM 010 Hapi_schemam_un_htcds\n\n"
echo hxltmcli "$Hapi_schemam_un_htcds" "$Hapi_schemam_un_htcds_RESULTATUM_TMX_LOCAL" --objectivum-TMX
hxltmcli "$Hapi_schemam_un_htcds" "$Hapi_schemam_un_htcds_RESULTATUM_TMX_LOCAL" --objectivum-TMX

printf "\n\n\n\tTESTUM 011 Hapi_schemam_un_htcds\n\n"
# echo hxltmcli "$Hapi_schemam_un_htcds" "$Hapi_schemam_un_htcds_RESULTATUM_BILINGUAL_HXL_CSV_LOCAL" --objectivum-CSV-HXL-XLIFF --fontem-linguam eng-Latn@en --objectivum-linguam por-Latn@pt
# hxltmcli "$Hapi_schemam_un_htcds" "$Hapi_schemam_un_htcds_RESULTATUM_BILINGUAL_HXL_CSV_LOCAL" --objectivum-CSV-HXL-XLIFF --fontem-linguam eng-Latn@en --objectivum-linguam por-Latn@pt
echo hxltmcli "$Hapi_schemam_un_htcds" "$Hapi_schemam_un_htcds_RESULTATUM_BILINGUAL_HXL_CSV_LOCAL" --objectivum-CSV-3--fontem-linguam eng-Latn@en --objectivum-linguam por-Latn@pt
hxltmcli "$Hapi_schemam_un_htcds" "$Hapi_schemam_un_htcds_RESULTATUM_BILINGUAL_HXL_CSV_LOCAL" --objectivum-CSV-3 --fontem-linguam eng-Latn@en --objectivum-linguam por-Latn@pt

printf "\n\n\n\tTESTUM 012 Hapi_schemam_un_htcds\n\n"
echo hxltmcli "$Hapi_schemam_un_htcds" "$Hapi_schemam_un_htcds_RESULTATUM_XLIFF_LOCAL" --objectivum-XLIFF --fontem-linguam eng-Latn@en --objectivum-linguam por-Latn@pt
hxltmcli "$Hapi_schemam_un_htcds" "$Hapi_schemam_un_htcds_RESULTATUM_XLIFF_LOCAL" --objectivum-XLIFF --fontem-linguam eng-Latn@en --objectivum-linguam por-Latn@pt

printf "\n\n\n\tTESTUM 013 --objectivum-formulam exemplum-linguam --objectivum-linguam por-Latn@pt \n\n"
echo 'hxltmcli schemam-un-htcds.tm.hxl.csv --objectivum-formulam formulam/exemplum-linguam.🗣️.json --objectivum-linguam por-Latn@pt > resultatum/formulam/exemplum-linguam.por-Latn.json'
hxltmcli schemam-un-htcds.tm.hxl.csv --objectivum-formulam formulam/exemplum-linguam.🗣️.json --objectivum-linguam por-Latn@pt > resultatum/formulam/exemplum-linguam.por-Latn.json

# https://en.wikipedia.org/wiki/Tocharian_languages
# https://www.reddit.com/r/linguisticshumor/comments/oynlth/not_sure_who_here_has_been_following_the/
printf "\n\n\n\tTESTUM 014 (Tocharian B) --objectivum-formulam exemplum-linguam --objectivum-linguam xto-Brah --auxilium-linguam xtb-Brah,lat-Latn,por-Latn,iir-Armi \n\n"
echo 'hxltmcli schemam-un-htcds.tm.hxl.csv --objectivum-formulam formulam/exemplum-linguam.🗣️.json --objectivum-linguam xto-Brah --auxilium-linguam xtb-Brah,lat-Latn,por-Latn,iir-Armi > resultatum/formulam/exemplum-linguam.xto-Brah~xtb-Brah,lat-Latn,por-Latn,iir-Armi.json'
hxltmcli schemam-un-htcds.tm.hxl.csv --objectivum-formulam formulam/exemplum-linguam.🗣️.json --objectivum-linguam xto-Brah --auxilium-linguam xtb-Brah,lat-Latn,por-Latn,iir-Armi > resultatum/formulam/exemplum-linguam.xto-Brah~xtb-Brah,lat-Latn,por-Latn,iir-Armi.json

# hxltmcli schemam-un-htcds.tm.hxl.csv --objectivum-formulam formulam/exemplum-linguam.🗣️.json --objectivum-linguam por-Latn@pt

# hxltmcli schemam-un-htcds.tm.hxl.csv resultatum/schemam-un-htcds.xlf --objectivum-XLIFF --fontem-linguam eng-Latn@en

# hxltmcli schemam-un-htcds.tm.hxl.csv --objectivum-formulam formulam/exemplum-linguam.🗣️.json --objectivum-linguam por-Latn@pt

# To revert only one file that keeps changing even with same input
# git checkout -- testum/hxltm/resultatum/hxltm-exemplum-linguam.tmx

# hxltmcli hxltm-exemplum-linguam.tm.hxl.csv --meta-archivum hxltm-exemplum-linguam.tmeta.yml
# hxltmcli hxltm-exemplum-linguam.tm.hxl.csv --meta-archivum hxltm-exemplum-linguam.tmeta.yml --objectivum-formulam formulam/exemplum-linguam.🗣️.json
# hxltmcli hxltm-exemplum-linguam.tm.hxl.csv --meta-archivum hxltm-exemplum-linguam.tmeta.yml --objectivum-formulam formulam/exemplum-linguam-tmeta-testum.🗣️.json
# hxltm-exemplum-linguam.tm.hxl.csv --tmeta-archivum hxltm-exemplum-linguam.tmeta.yml --objectivum-formulam formulam/exemplum-linguam-tmeta-testum.🗣️.json

exit 0
